pilot/pkg/security/authz/builder/builder_test.go

```diff
@@ -114,6 +129,12 @@ func TestGenerator_GenerateHTTP(t *testing.T) {
 			input: "path15-in.yaml",
 			want:  []string{"path15-out.yaml"},
 		},
+		{
+			name:       "action-custom-grpc-provider-no-namespace",
```

Comment:
should have a test case where multiple services are found and returning error?

Reply:
I'm not sure how useful to test the invalid input for this case, we don't test other invalid inputs as well but we cover all the valid inputs, that should be good enough.

Reply:
well, at least then the "else" branch code is not covered. someone later can remove entire else branch of `len(namespaces) > 1` to a incorrect impl without CI complaining at all.

Reply:
that just means a less useful error message but it doesn't affect the functionality, we cover the normal code path but currently doesn't cover all the error path